src/PriorityQueue/PHP81Implementation.php

```diff
@@ -0,0 +1,403 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Stdlib\PriorityQueue;
+
+use Countable;
+use IteratorAggregate;
+use Laminas\Stdlib\Exception;
+use Laminas\Stdlib\SplPriorityQueue;
+use Serializable;
+use Traversable;
+use UnexpectedValueException;
+
+use function array_key_exists;
+use function array_map;
+use function count;
+use function get_class;
+use function gettype;
+use function is_array;
+use function is_object;
+use function serialize;
+use function sprintf;
+use function unserialize;
+
+/**
+ * Re-usable, serializable priority queue implementation
+ *
+ * SplPriorityQueue acts as a heap; on iteration, each item is removed from the
+ * queue. If you wish to re-use such a queue, you need to clone it first. This
+ * makes for some interesting issues if you wish to delete items from the queue,
+ * or, as already stated, iterate over it multiple times.
+ *
+ * This class aggregates items for the queue itself, but also composes an
+ * "inner" iterator in the form of an SplPriorityQueue object for performing
+ * the actual iteration.
+ */
+class PHP81Implementation implements Countable, IteratorAggregate, Serializable
```

Comment:
Should be `@internal` IMO: the implementation should not be part of the public API